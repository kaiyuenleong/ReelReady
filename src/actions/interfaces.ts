interface Registration {
  name: string;
  email: string;
  password: string;
  confirmPassword: string;
  image: string;
}

export { Registration };