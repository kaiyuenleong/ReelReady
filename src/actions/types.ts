// Login Types
export const EMAIL_CHANGED = 'email_changed';
export const PASSWORD_CHANGED = 'password_changed';
export const LOGIN_USER_SUCCESS = 'login_user_success';
export const LOGIN_USER_FAIL = 'login_user_fail';
export const LOGIN_USER = 'login_user';

// Registration Types
export const NEW_NAME_CHANGED = 'new_name_changed';
export const NEW_EMAIL_CHANGED = 'new_email_changed';
export const NEW_PASSWORD_CHANGED = 'new_password_changed';
export const NEW_CONFIRM_PASSWORD_CHANGED = 'new_confirm_password_changed';
export const REGISTER_USER_SUCCESS = 'register_user_success';
export const REGISTER_USER_FAIL = 'register_user_fail';
export const REGISTER_USER = 'register_user';
