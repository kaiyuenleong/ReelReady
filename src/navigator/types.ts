import { RouteProp } from "@react-navigation/native";
import { StackNavigationProp } from "@react-navigation/stack";

type RootStackParamList = {
  Login: {};
  Registration: {};
  ForgotPassword: {};

  Home: {};
}

type LoginScreenRouteProp = RouteProp<RootStackParamList, 'Login'>;
type LoginScreenNavigationProp = StackNavigationProp<RootStackParamList, 'Login'>;

type RegistrationScreenRouteProp = RouteProp<RootStackParamList, 'Registration'>;
type RegistrationScreenNavigationProp = StackNavigationProp<RootStackParamList, 'Registration'>;

type ForgotPasswordScreenRouteProp = RouteProp<RootStackParamList, 'ForgotPassword'>;
type ForgotPasswordScreenNavigationProp = StackNavigationProp<RootStackParamList, 'ForgotPassword'>;

type HomeScreenRouteProp = RouteProp<RootStackParamList, 'Home'>;
type HomeScreenNavigationProp = StackNavigationProp<RootStackParamList, 'Home'>;

export { 
  RootStackParamList, 
  LoginScreenRouteProp, 
  LoginScreenNavigationProp,
  RegistrationScreenRouteProp,
  RegistrationScreenNavigationProp,
  ForgotPasswordScreenRouteProp,
  ForgotPasswordScreenNavigationProp,

  HomeScreenRouteProp,
  HomeScreenNavigationProp
};