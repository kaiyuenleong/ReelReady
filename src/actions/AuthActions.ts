import firebase from "firebase";
import {
  EMAIL_CHANGED,
  PASSWORD_CHANGED,
  LOGIN_USER_SUCCESS,
  LOGIN_USER_FAIL,
  LOGIN_USER
} from "./types";

const emailChanged = (text: string) => {
  return {
    type: EMAIL_CHANGED,
    payload: text
  }
}

const passwordChanged = (text: string) => {
  return {
    type: PASSWORD_CHANGED,
    payload: text
  }
}

const loginUser = ({ email, password }: any) => {
  return (dispatch: any) => {
    dispatch({ type: LOGIN_USER });

    // Call API to login user here
  }
}

const loginUserFail = (dispatch: any) => {
  dispatch({ type: LOGIN_USER_FAIL });
}

const loginUserSuccess = (dispatch: any, user: any) => {
  dispatch({
    type: LOGIN_USER_SUCCESS,
    payload: user
  });
}

export {
  emailChanged, 
  passwordChanged,
  loginUser,
  loginUserFail,
  loginUserSuccess
}