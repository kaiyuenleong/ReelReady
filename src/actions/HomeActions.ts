import {
  HOME_MOUNTED
} from "./types";

const homeMounted = (data: any) => {
  return (dispatch: any) => {
    dispatch({ type: HOME_MOUNTED, payload: data });
  }
}

export {
  homeMounted
}