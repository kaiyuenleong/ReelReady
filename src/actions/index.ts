export * from "./LoginActions";
export * from "./RegistrationActions";
export * from "./HomeActions";