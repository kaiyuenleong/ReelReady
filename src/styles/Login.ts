import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  contentContainer: {
    paddingLeft: 30,
    paddingRight: 30,
    paddingTop: 30,
    paddingBottom: 30,
    flex: 1
  },
  image: { 
    height: 250, 
    width: 250
  },
  imageContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  centerContainer: {
    flex: 1
  },
  textContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "flex-start",
    flex: 1
  },
  commonContainer: {
    flex: 1,
    justifyContent: "space-around"
  },
  fieldsContainer: {
    flex: 3,
    justifyContent: "space-around"
  },
  registrationText: {
    textAlign: 'center',
    color: '#FFFFFF',
    fontSize: 16,
    fontWeight: '200',
    marginTop: 20,
    letterSpacing: 0.2
  },
  registrationSubtext: {
    color: '#9CFFEE',
    fontWeight: "bold"
  },
  forgotPasswordText: {
    textAlign: 'right',
    color: '#D9D9D9',
    fontSize: 14,
    fontWeight: '200',
    letterSpacing: 0.2
  },
  errorText: {
    fontSize: 14,
    color: 'red'
  }
});