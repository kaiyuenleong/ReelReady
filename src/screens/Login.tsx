import React, { Component } from "react";
import { Text, Image, View, TouchableOpacity } from "react-native";
import { LoginScreenRouteProp, LoginScreenNavigationProp } from "../navigator/types";
import { connect } from "react-redux";
import { emailChanged, passwordChanged, loginUser } from "../actions";
import { Gradient, Button, Input } from "../components/common";
import styles from "../styles/Login";

interface LoginAction {
	type: string;
	payload: string;
}

interface LoginProps {
	email: string;
	password: string;
	error: string;
	loading: boolean;

	emailChanged: (text: string) => LoginAction;
	passwordChanged: (text: string) => LoginAction;
	loginUser: any;
	
	route: LoginScreenRouteProp;
	navigation: LoginScreenNavigationProp;
}

class Login extends Component<LoginProps> {

	register = () => {
		this.props.navigation.navigate('Registration');
	}

	forgotPassword = () => {
		this.props.navigation.navigate('ForgotPassword');
	}

	onEmailChange = (text: string) => {
		this.props.emailChanged(text);
	}

	onPasswordChange = (text: string) => {
		this.props.passwordChanged(text);
	}

	onButtonPress = () => {
		const { email, password } = this.props;
		this.props.loginUser({ email, password });
	}

	// Need to change button to loading circle when checking authentication?
	renderButton() {
		return (
			<Button onPress={this.onButtonPress}>SIGN IN</Button>
		)
	}

	render() {
		return (
			<Gradient>
				<View style={styles.contentContainer}>
					<View style={styles.imageContainer}>
						<Image
							source={require('../../assets/images/logo_w_text.png')}
							resizeMode="contain"
							style={styles.image}
						/>
					</View>
					<View style={styles.inputContainer}>
						<Input
							placeholder="EMAIL"
							onChangeText={this.onEmailChange}
							value={this.props.email}
						/>
						<Input
							secureTextEntry
							placeholder="PASSWORD"
							onChangeText={this.onPasswordChange}
							value={this.props.password}
						/>
					</View>
					<View style={styles.buttonContainer}>
						<View style={styles.textContainer}>
							<Text style={styles.errorText}>
								{this.props.error}
							</Text>
							<TouchableOpacity onPress={this.forgotPassword}>
								<Text style={styles.forgotPasswordText}>
									Forgot password?
								</Text>
							</TouchableOpacity>
						</View>
						{this.renderButton()}
						<TouchableOpacity onPress={this.register}>
							<Text style={styles.registrationText}>New User? Create an Account</Text>
						</TouchableOpacity>
					</View>
				</View>
			</Gradient>
		)
	}
}

const mapStateToProps = ({ login }: any) => {
	const { email, password, error, loading } = login;
	return { email, password, error, loading };
}

export default connect(mapStateToProps, { emailChanged, passwordChanged, loginUser })(Login);