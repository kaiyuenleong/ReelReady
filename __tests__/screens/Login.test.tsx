import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import * as types from "../../src/actions/types";
import configureStore from "redux-mock-store";
import { render, cleanup, fireEvent } from "react-native-testing-library";
import Login from "../../src/screens/Login";

afterEach(cleanup);
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('login', () => {
  const state = { login: { email: "", password: "", error: "", loading: false }}

  it('renders correctly', () => {
    const store = mockStore(state);
    const tree = render(<Provider store={store}><Login /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should dispatch the login action', () => {
    const store = mockStore(state);
    const tree = render(<Provider store={store}><Login /></Provider>);
    const signInButtonComponent = tree.getByText("SIGN IN");
    
    fireEvent(signInButtonComponent, "press");

    const actions = store.getActions();
    expect(actions.length).toBe(1);
    expect(actions[0].type).toEqual(types.LOGIN_USER);
  })

  it('should dispatch the email changed action', () => {
    const store = mockStore(state);
    const tree = render(<Provider store={store}><Login /></Provider>);
    const emailInputComponent = tree.getByPlaceholder("EMAIL");
    
    fireEvent.changeText(emailInputComponent, "testing@testing.com");

    const actions = store.getActions();
    expect(actions.length).toBe(1);
    expect(actions[0].type).toEqual(types.EMAIL_CHANGED);
  })

  it('should dispatch the password changed action', () => {
    const store = mockStore(state);
    const tree = render(<Provider store={store}><Login /></Provider>);
    const passwordInputComponent = tree.getByPlaceholder("PASSWORD");
    
    fireEvent.changeText(passwordInputComponent, "testingpassword");

    const actions = store.getActions();
    expect(actions.length).toBe(1);
    expect(actions[0].type).toEqual(types.PASSWORD_CHANGED);
  })
})