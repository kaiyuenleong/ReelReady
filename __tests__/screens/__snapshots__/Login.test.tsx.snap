// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4289198296,
          4283231449,
          4279833944,
          4279304248,
        ],
        "endPoint": undefined,
        "locations": Array [
          0,
          0.1,
          0.8,
          1,
        ],
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "paddingBottom": 30,
            "paddingLeft": 30,
            "paddingRight": 30,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": 250,
                "width": 250,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 3,
                  "justifyContent": "space-around",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCompleteType="off"
                  autoCorrect={false}
                  onChangeText={[Function]}
                  placeholder="EMAIL"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "borderBottomColor": "rgba(217, 217, 217, 1)",
                      "borderBottomWidth": 0.5,
                      "color": "#FFFFFF",
                      "flex": 1,
                      "fontSize": 15,
                      "padding": 2,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCompleteType="off"
                  autoCorrect={false}
                  onChangeText={[Function]}
                  placeholder="PASSWORD"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Object {
                      "borderBottomColor": "rgba(217, 217, 217, 1)",
                      "borderBottomWidth": 0.5,
                      "color": "#FFFFFF",
                      "flex": 1,
                      "fontSize": 15,
                      "padding": 2,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "red",
                    "fontSize": 14,
                  }
                }
              >
                
              </Text>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#D9D9D9",
                      "fontSize": 14,
                      "fontWeight": "200",
                      "letterSpacing": 0.2,
                      "textAlign": "right",
                    }
                  }
                >
                  Forgot password?
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "rgba(255, 255, 255, 0.4)",
                  "borderColor": "rgba(255, 255, 255, 0.4)",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "height": 60,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "letterSpacing": 0.5,
                    "textAlign": "center",
                  }
                }
              >
                SIGN IN
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                    "fontWeight": "200",
                    "letterSpacing": 0.2,
                    "marginTop": 20,
                    "textAlign": "center",
                  }
                }
              >
                New user?
                <Text
                  style={
                    Object {
                      "color": "#9CFFEE",
                      "fontWeight": "bold",
                    }
                  }
                >
                   Create an account
                </Text>
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
